logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -342,6 +342,8 @@ def process_parameter_value(value, config_settings)
         # Empty lists are converted to nils
         return true, [] if value.empty?
 
+        return validate_value(value, :uri_list) if config_val == :uri
```

Comment:
if I read correctly, removing this line would allow a list of strings, each containing an environment variable, to be expanded, or even a fixed list + an environment variable?
I can see some farfetched scenarios that would benefit from this flexibility, but another argument is that we maybe don't need this special case

Reply:
I found that it is simpler to support it this way, and that constraining it to a single entry that produced many was much more special-case-y in implementation.

 - When we have `:list => true`, each entry is validated/coerced with its validator to _exactly one_ result, so the validator cannot be used to split an entry into many without _all_ validators being updated to produce one-or-more results (this is especially hard because plugins patch in their own validation in various ways, including redefining `config_init`, `validate_value`, and others)
 - publicly introducing a new validation type (like the _private_ `:uri_list` here) to be consumed directly by plugins produces a wall where those plugins cannot work on versions of Logstash that didn't provide the validator.